src/extensions/MarketCreation.sol

```diff
@@ -0,0 +1,34 @@
+pragma solidity ^0.4.13;
+
+
+// FIXME: Remove this stub once the contract is implemented in Solidity
+contract Market {
+    function stub();
+}
+
+
+// FIXME: Remove this stub once the contract is implemented in Solidity
+contract ReportingWindow {
+    function createNewMarket(int256 _endTime, int256 _numOutcomes, int256 _payoutDenominator, int256 _feePerEthInWei, address _denominationToken, address _sender, int256 _minDisplayPrice, int256 _maxDisplayPrice, address _automatedReporterAddress, int256 _topic) payable returns (Market);
+}
+
+
+// FIXME: Remove this stub once the contract is implemented in Solidity
+contract Branch {
+    function getReportingWindowByMarketEndTime(int256 _endTime, int256 _hasAutomatedReporter) returns (ReportingWindow);
+}
+
+
+contract MarketCreation {
+    function createScalarMarket(Branch _branch, int256 _endTime, int256 _feePerEthInWei, address _denominationToken, int256 _minDisplayPrice, int256 _maxDisplayPrice, address _automatedReporterAddress, int256 _topic) payable public returns (Market _market) {
+        ReportingWindow _reportingWindow = _branch.getReportingWindowByMarketEndTime(_endTime, _automatedReporterAddress != 0 ? 1 : 0);
```

Comment:
Have we already had the `var` discussion?  If so tell me what our conclusion was.  I'm personally a fan of `var`, _especially_ when the type is obvious (like this).  I believe the argument against `var` is the fact that it chooses the most constrained type it can, which means `uint8` if you do `var i = 0`.  Knowing this puts me a bit on the fence with Solidity, so I am curious to hear if others have strong opinions.  Also, because editors right now lack strong inference support, unlike Kotlin/C# it isn't trivial to figure out what the inferred type of something is so when it is non-obvious from context, having the explicit type name can be valuable.